### PR TITLE
documentation: Update for Charm 8.0.0

### PR DESCRIPTION
**Pull request by ericjbohm**
Updates to release oriented documentation for Charm 8.0.0

### CODE REVIEW

**Review by ericjbohm — COMMENTED** [review]
Add bullet with brief summary of license change

**Comment by ericjbohm** [comment]
It looks like choosing a branch with the full version name has triggered some branch locking which makes modification unusually difficult.  Not sure how to fix that.

**Comment by ericjbohm** [comment]
> It looks like choosing a branch with the full version name has triggered some branch locking which makes modification unusually difficult. Not sure how to fix that.

Revised branch protection rules so we can revise 7.1.0.  Will restore restrictions when candidate is ready.

**Comment by stwhite91** [comment]
Some other suggestions for inclusion in the release notes:

- Added support for NVIDIA's nvc/nvfortran compilers and Intel's new icx/ifx compilers.
- Fixed a bug in location management when doing dynamic insertion and deletion of chare array elements.
- Deprecated the "atomic" keyword in SDAG code (in favor of "serial").
- Added support for TLSglobals on Apple ARM systems and the ability to disable TLSglobals support at build-time.
- Performance optimizations for node group broadcasts.
- Added CMI-SHMEM module for optimizing small/medium-sized messages between processes on the same host.
- Fixed demand creation via ckcallback::send and enabled passing options to ckcallback::send.

File: CHANGES [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+
+- The License has changed to Apache 2.0 with LLVM exception.  This
+  change to a popular Open Source license is intended to simplify use,
+  collaboration, and greater community involvement in the development
+  of Charm++.
+
+- The CkIO library has been enhanced to support both input and output.
+ <insert more details here>
+
+- The OFI layer has been extended to support the CXI (Cassini)
```
Comment (stwhite91) [inline CHANGES:19]: I would add that explicitly adding cxi to the build command is not necessary on Cray systems. Also, I would hope that we print something saying that cxi is enabled at startup when it is on.

Comment (ericjbohm) [reply CHANGES:24]: Note added.  PR https://github.com/UIUC-PPL/charm/pull/3810 adds output at startup when CXI is in use.

File: CMakeLists.txt [hunk 2]
```diff
@@ -30,7 +30,7 @@ if(CMAKE_VERSION VERSION_GREATER 3.12 OR CMAKE_VERSION VERSION_EQUAL 3.12)
 endif()
 
 # Fortran is optional, so don't include it in LANGUAGES here
-project(Charm++ LANGUAGES CXX C ASM VERSION 7.0.0)
+project(Charm++ LANGUAGES CXX C ASM VERSION 7.1.0)
```
Comment (matthiasdiener) [inline CMakeLists.txt:33]:
```suggestion
project(Charm++ LANGUAGES CXX C ASM VERSION 8.0.0)
```

File: README.md [hunk 3]
```diff
@@ -365,24 +365,26 @@ please create an issue or discussion on our GitHub, https://github.com/UIUC-PPL/
...
-
-Copyright (C) 1989-2023 Regents of the University of Illinois
+Seonmyeong Bak, Karthik Senthil, Juan Galvez, Michael Robson,
+Raghavendra Kanakagiri, Venkatasubrahmanian Narayanan. Nitin Bhat, and
+Justin Szaday. Current developers include: Eric Bohm, Ronak Buch, Eric
+Mikida, Sam White, Kavitha Chandrasekar, Jaemin Choi, Matthias Diener,
+Evan Ramos, Zane Fink, Pathikrit Ghosh, Maya Taylor, Aditya Bhosale,
+Mathew Jacob, Tom Vander Aa, and Thomas Quinn.
+
+Copyright (C) 1989-2024 Regents of the University of Illinois
```
Comment (matthiasdiener) [inline README.md:390]: Isn't it 

```suggestion
Copyright (C) 1989-2024 University of Illinois Board of Trustees
```

?

Comment (ericjbohm) [reply README.md:390]: Those suggestions are sound.  However, as this PR was made for the (now defunct) idea of 7.1.0 from that branch, I think we need to make a whole new PR from the 8.0 branch for this.

**Comment by ericjbohm** [comment]
See https://github.com/UIUC-PPL/charm/pull/3816 which includes these changes, but from 8.0 branching instead of 7.1

**Pull request closed without merging**